### PR TITLE
Update UG

### PR DESCRIPTION
**Pull request by bharathcs**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -11,16 +11,17 @@ title: Developer Guide
...
+This guide covers the architecture, implementation and design choices in DonnaFin to give the reader a clear picture
+of the technical details and the inner workings of DonnaFin.
 
 ### 1.2 Target Audience
-This developer guide is aimed at developers and advanced users of DonnaFin
-* Developers: anyone who wishes to upgrade DonnaFin to support more functions.
-* Advanced Users: Financial Advisors who want to better understand the features that DonnaFin Provides.
+
+* Developers: anyone who wishes to dive into this codebase to improve or extend on DonnaFin.
+* Advanced Users: financial advisors who want to better understand the features that DonnaFin Provides.
```
Comment (sheehui) [inline docs/DeveloperGuide.md:21]: provides

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -81,7 +92,7 @@ Given below is a quick overview of main components and how they interact with ea
 **Main components of the architecture**
 
 **`Main`** has two classes called [`Main`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/Main.java) and [`MainApp`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/MainApp.java). It is responsible for,
```
Comment (sheehui) [inline docs/DeveloperGuide.md:94]: we should link to our repo instead

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -81,7 +92,7 @@ Given below is a quick overview of main components and how they interact with ea
...
-* At app launch: Initializes the components in the correct sequence, and connects them up with each other.
+* At app launch: Initializes the components in the correct sequence, and connects them up with each other
```
Comment (sheehui) [inline docs/DeveloperGuide.md:95]: no need 'up' i feel

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -96,14 +107,28 @@ The rest of the App consists of four components.
...
+:bulb: **Tip:** There are some notable exceptions to the `Facade` pattern. For example, for the `Ui` component, 
+`AttributePanel`, `PersonCard` and `ClientPanel`. Arguably the better implementation would be to restrict all the 
+interactions between components to the facades, but a design decision was made to prioritize a simple implementation of
+critical features over adhering to the design pattern.
+</div>
+
+
 #### 4.1.1 UI component
 
 [`Ui.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/Ui.java) specifies the API of this component.
```
Comment (sheehui) [inline docs/DeveloperGuide.md:134]: same thing here

**Review by sheehui — COMMENTED** [review]
v nice and good idea!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (bharathcs) [inline docs/DeveloperGuide.md:134]:
```suggestion
[`Ui.java`](https://github.com/AY2122S1-CS2103T-W16-1/tp/tree/master/src/main/java/donnafin/ui/Ui.java) specifies the API of this component.
```

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (bharathcs) [inline docs/DeveloperGuide.md:94]:
```suggestion
**`Main`** has two classes called [`Main`](https://github.com/AY2122S1-CS2103T-W16-1/tp/tree/master/src/main/java/donnafin/Main.java) and [`MainApp`](https://github.com/AY2122S1-CS2103T-W16-1/tp/tree/master/src/main/java/donnafin/MainApp.java). It is responsible for,
```

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (bharathcs) [reply docs/DeveloperGuide.md:95]:
```suggestion
* At app launch: Initializes the components in the correct sequence, and connects them with each other
```

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (bharathcs) [reply docs/DeveloperGuide.md:21]:
```suggestion
* Advanced Users: financial advisors who want to better understand the features that DonnaFin provides.
```

**Comment by bharathcs** [comment]
> v nice and good idea!

Nice catches!

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (bharathcs) [reply docs/DeveloperGuide.md]: Done

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (bharathcs) [reply docs/DeveloperGuide.md]: Done

**Review by sheehui — APPROVED** [review]
aight gtg

**Pull request merged**